### DISCUSSION TITLE
Msys2 and VS Code task

### DISCUSSION THREAD

**Discussion by A2va** [msys2/MINGW-packages#7674] [General]
I try to implement task for build a c file with msys2 and vs code. 

First I tried to change the default shell with a settings.json inside my .vscode folder
```json
{
    "terminal.integrated.shell.windows": "D:/compiler/msys64/usr/bin/bash.exe",
    "terminal.integrated.env.windows": {
        "MSYSTEM": "MINGW32",
        "MSYS2_PATH_TYPE": "inherit"
    }
}
```
```json
{
    "terminal.integrated.shell.windows": "D:/compiler/msys64/msys2_shell.cmd",
    "terminal.integrated.shellArgs.windows": [
        "-mingw32", "-defterm", "-no-start", "-here", "-full-path"
    ]
}
```
 The two modification of settings above are worked. So I tried the same idea inside the tasks.json and my task are:
```json
{
            "label": "Build",
            "type": "shell",
            "isBackground": false,
            "command": "gcc",
            "args": [
                "-g",
                "-Wall",
                "${workspaceFolder}/hello_world.c",
                "-o",
                "out.exe"
            ],
            "options": {
                "cwd": "${workspaceFolder}",
                "shell": {
                    "executable": "D:/compiler/msys64/usr/bin/bash.exe"
                },
                "env": {
                    "MSYSTEM": "MINGW32",
                    "MSYS2_PATH_TYPE": "inherit"
                }
            },
            "problemMatcher": {
                "base": "$gcc",
                "fileLocation": "absolute"
            }
        }
```
But vs code tried to execute gcc into the workspace folder so What would be the solution ?

**Comment by 1480c1** [comment]
Not entirely sure if you mean you don't want gcc to run in the workspace folder or not since you don't actually present what the problem is but the conclusion, but one variable you might want to look into is `CHERE_INVOKING`

**Reply by 1480c1** [reply]
Also, not sure if this is the kind of thing for discussions and more like an issue, depending on if you are asking about an issue with vscode and msys2 or just wanting to discuss the best ways to use the two together

**Reply by A2va** [reply]
> Not entirely sure if you mean you don't want gcc to run in the workspace folder or not since you don't actually present what the problem is but the conclusion, but one variable you might want to look into is `CHERE_INVOKING`

I modified the task.json by adding the chere_invoking : 

```json
"env": {
           "MSYSTEM": "MINGW32",
           "MSYS2_PATH_TYPE": "inherit",
           "CHERE_INVOKING":1
 }
```
It doesn't build the file, the bash path are in /usr/bin/bash 

Also if I remove the shell executable in task.json and i put in settings.json the same configuration and it works. 

Now my tasks.json are this:
```json
 {
            "label": "Build",
            "type": "shell",
            "isBackground": false,
            "command": "gcc",
            "args": [
                "-g",
                "hello_world.c",
                "-o",
                "out.exe"
              ],
            "options": {
                "cwd": "${workspaceFolder}",
            },
            "problemMatcher": {
                "base": "$gcc",
                "fileLocation": "absolute"
            }
        }
```
settings.json:
```json
"terminal.integrated.shell.windows": "D:/compiler/msys64/usr/bin/bash.exe",
"terminal.integrated.env.windows": {
        "MSYSTEM": "MINGW32",
        "MSYS2_PATH_TYPE": "inherit",
        "CHERE_INVOKING":"1"
    }
````
Indeed now it seems to me more like a bug in the definition of a shell in the tasks.json different from the default one.

**Comment by jamestansx** [comment]
Referring to your last thread in the first comment, you are able to build the c file. I was wondering if you are able to debug the file using gdb (MSYS2) in vscode?

**Reply by A2va** [reply]
Yes it's possible.
Here an example for the launch.json:
```json
  {
            "name": "(gdb) Launch",
            "type": "cppdbg",
            "request": "launch",
            "program": "enter program name, for example ${workspaceFolder}/a.exe",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": false,
            "MIMode": "gdb",
            "miDebuggerPath": "/path/to/gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ]
        }
```
Don't forget to install C/C++ extension.
